datafusion/src/logical_plan/dfschema.rs

```diff
@@ -160,7 +160,13 @@ impl DFSchema {
                 // field to lookup is qualified.
                 // current field is qualified and not shared between relations, compare both
                 // qualifer and name.
-                (Some(q), Some(field_q)) => q == field_q && field.name() == name,
+                (Some(q), Some(field_q)) => {
```

Comment:
nitpick, i think it would be good to check for `field.name() == name` first and do an early return if it doesn't match to avoid the subsequent field parsing/matching.

Reply:
ok no problem.